### PR TITLE
Use UTC time for reservation interactions.

### PR DESCRIPTION
**Pull request by wcourtney**

### CODE REVIEW

File: cirq/google/engine/engine_processor.py [hunk 1]
```diff
@@ -223,7 +224,7 @@ def remove_reservation(self, reservation_id: str):
             raise ValueError('Cannot determine freeze_schedule from processor.'
                              'Call _cancel_reservation or _delete_reservation.')
         secs_until = (reservation.start_time.seconds -
-                      int(datetime.datetime.now().timestamp()))
+                      int(datetime.datetime.now(tz=utc).timestamp().timestamp()))
```
Comment (dstrain115) [inline cirq/google/engine/engine_processor.py]: Why are there two timestamps() here?

**Comment by dstrain115** [comment]
I am wondering why this matters.  It should be turned into epoch timestamps everywhere which should be timezone independent.

File: cirq/google/engine/engine_processor.py (hunk 1 above)
Comment (wcourtney) [reply cirq/google/engine/engine_processor.py]: Apparently I hadn't pushed the fix for this. \Done.

**Comment by wcourtney** [comment]
> I am wondering why this matters. It should be turned into epoch timestamps everywhere which should be timezone independent.

The epoch timestamps are created from localized dates. Specifically, the PR is a bugfix for operation In colab, where datetime.now() provides a timestamp that is 7 hours later than running locally in PT. I've resolved that on the client side for the parameterized values, remove_timeslot tries to cancel reservations that it should delete and fails.

**Pull request merged**